pkg/util/kubeobjects/internal/query/query.go

```diff
@@ -0,0 +1,226 @@
+package query
+
+import (
+	"context"
+	goerrors "errors"
+	"reflect"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/scheme"
+	"github.com/Dynatrace/dynatrace-operator/pkg/logd"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type Generic[T client.Object, L client.ObjectList] struct {
+	Target      T
+	ListTarget  L
+	ToList      func(L) []T
+	IsEqual     func(T, T) bool
+	IsImmutable func(T, T) bool
```

Comment:
Would rather rename it to something like `RecreationRequired`. 
During the review I always had to think about what you mean with IsImmutable. Maybe `IsImmutableFieldChanged` would be ok to.

Reply:
done